### PR TITLE
Warnings fixes + optimizations + floating-point consistency + clang-format

### PR DESCRIPTION
**Pull request by jrmadsen**
- Warning fixes
    - Fixed sign comparison warnings in src/morph.c
- Optimizations in src/utils.c via SIMD instructions
    - enable with `export CFLAGS="-fopenmp"` before build
        - this compiler flags will eventually be tested for availability and automatically applied when CMake build system is implemented
    - recorded speed-up on Cori-KNL, Cori-Haswell, Edison: ~1.25x
    - Optimized functions
        - calc_quadrant
        - calc_coords
        - trim_coords
        - sort_intersections
        - calc_dist
- floating-point consistency
    - replaced div by 2 with multiply by 0.5f (faster)
    - replaced double constants with float constants
    - replaced fmod with fmodf
- clang-format
    - created .clang-format file (requires clang-format v6.0+)
        - run: `clang-format -i $(find . -type f | egrep '\.h$|\.c$')` to apply formatting in-place

### CODE REVIEW

**Comment by dgursoy** [comment]
Conflict due to #366 merge.

File: src/utils.c [hunk 1]
```diff
@@ -1,342 +1,459 @@
...
+    for(int i = 0; i <= rz; ++i)
     {
-        gridy[i] = -rz/2.+i;
+        gridy[i] = -rz * 0.5f + i;
     }
 
-    *mov = ((float)num_pixels-1)/2.0-center;
-    if(*mov-floor(*mov) < 0.01) {
-        *mov += 0.01;
+    *mov = ((float) num_pixels - 1) * 0.50f - center;
```
Comment (carterbox) [inline src/utils.c]: There's a trailing zero here that's not necessary.

Comment (jrmadsen) [reply src/utils.c]: Fair point about separate commit for formatting. Normally I would have but wasn't paying attention to that when I rebased bc I was more concerned about ensuring any the MPI stuff wasn't included. The only C code of import affected by the optimizations was the functions listed in utils.c. For the algorithms it was FP consistency

**Comment by jrmadsen** [comment]
If you want to deny this PR, I can re-do the rebase so the clang-format is separate.

**Comment by carterbox** [comment]
That's not necessary. I already read through it all. I'm only requesting removal of that trailing zero.

**Pull request merged**